src/typing.d.ts

```diff
@@ -0,0 +1,197 @@
+     */
+    setext?: boolean,
+    
+    /**
+     * Compile ATX headings with the same amount of closing hashes as opening hashes (default: false).
+     */
+    closeAtx?: boolean,
+    
+    /**
+     * Create tables without fences: initial and final pipes (default: false).
+     */
+    looseTable?: boolean,
+    
+    /**
+     * Create tables without spacing between pipes and content (default: true).
+     */
+    spacedTable?: boolean,
+    
+    /**
+     * Create tables without spacing between pipes and content (default: true).
+     */
+    paddedTable?: boolean,
+    
+    /**
+     * Create tables with padding in each cell so that they are the same size (default: true).
+     */
+    stringLength?: boolean,
+    
+    /** 
+     * Function passed to markdown-table to detect the length of a table cell (default: s => s.length).
+     */
+    fence?: (fence: string) => number,
+    
+    /**
+     * Fence marker to use for code blocks (default: '`').
+     */
+    fences?: '~' | '`',
+    
+    /**
+     * Stringify code blocks without language with fences (default: false).
+     */
+    bullet?: boolean,
+    
+    /**
+     * Bullet marker to use for unordered list items (default: '-').
+     */
+    listItemIndent?: '-' | '*' | '+',
+    
+    /**
+     * How to indent the content from list items (default: 'tab').
+     *  
+     * 'tab': use tab stops (4 spaces)
+     * '1': use one space
+     * 'mixed': use 1 for tight and tab for loose list items
+     */
+    incrementListMarker?: 'tab' | 'mixed' | '1',
+    
+    /**
+     * Whether to increment ordered list item bullets (default: true).
+     */
+    rule?: boolean,
+    
+    /**
+     * Marker to use for thematic breaks / horizontal rules (default: '*').
+     */
+    ruleRepetition?: '-' | '*' | '_',
+    
+    /**
+     * Number of markers to use for thematic breaks / horizontal rules (default: 3). Should be 3 or more.
+     */
+    ruleSpaces?: number,
+    
+    /**
+     *   Whether to pad thematic break (horizontal rule) markers with spaces (default true).
+     */
+    strong?: boolean,
+    
+    /**
+     *  Marker to use for importance (default '*').
+     */
+    emphasis?: '_' | '*'
+  }
+}
+
+declare module 'unified' {
+  import { ToVFile, VFile } from "vfile";
+
+  export type Plugin = Function | Array<[Function, any]>
+  export type Node = any
+
+  export interface Processor extends ProcessorFunc{
+    use(plugin: Plugin ,options?: any): this
+    parse(value: ToVFile): Node,
+    stringify(node: Node, file?: ToVFile): string,
+    run(node: Node, file?: ToVFile): Promise<Node>,
+    runSync(node: Node, file?: ToVFile): Node,
+    process(file: ToVFile): Promise<VFile>,
+    processSync(file: ToVFile): VFile,
+    data(key: string): any,
+    data(key: string, value: any): this,
+    freeze(): Processor
+  }
+  type ProcessorFunc = () => Processor
+  
+  const unified: Processor
+
+  export default unified
+}
+
+declare module 'vfile' {
```

Comment:
FYI, It looks like for the `vfile` module, there is actually a typings available - you can `npm install @types/vfile` and then you won't need to include the `declare module 'vfile'` block.

Reply:
Thanks